### PR TITLE
Handle capture-related exceptions in request-body gracefully

### PR DESCRIPTION
**Pull request by Altai-man**
It turns out ACCEPTS could throw an exception instead of silently
returning False when it cannot unpack something.

It is an ambiguous error we should not show to the client, so
better hide it behind a 400, but in fact it can be a developer's
mis-understanding somewhere as well, so report them some info
that can be useful in debugging.

Tries to fix https://github.com/croservices/cro-http/issues/132

### CODE REVIEW

**Comment by jnthn** [comment]
> It turns out ACCEPTS could throw an exception instead of silently
> returning False when it cannot unpack something.

Do you have an example of where this happens? Either:

1. It's a Rakudo bug, in which case we should fix it there
2. It really wants to be a 500 (for example, if one writes a subset type that throws, that's a bug the programmer should fix)

**Comment by jnthn** [comment]
Oh, duh, I see it now. Hm.

**Comment by Altai-man** [comment]
The example code is in the ticket attached.
I discussed this behavior with lizmat IIRC, but thought it's better we patch this than rakudo.

File: lib/Cro/HTTP/Router.pm6 [hunk 1]
```diff
@@ -27,6 +27,18 @@ class X::Cro::HTTP::Router::NoRequestBodyMatch is Exception {
     }
 }
 
+class X::Cro::HTTP::Router::ConfusedCapture is Exception {
+    has $.body;
+    has $.content-type;
+    has $.signature;
+
+    method message() {
+        "The message body was parsed into '$!body.raku()' of type ($!body.^name()), " ~
```
Comment (jnthn) [inline lib/Cro/HTTP/Router.pm6:36]: I think `$!body.raku` could end up huge; I think maybe let's:

1. Move it to the end after of the message
2. Length limit it (500 chars or so) and add "..."

**Review by jnthn — COMMENTED** [review]
I'm fine with this, just needs conflict fixing and a minor tweak...

**Review by jnthn — APPROVED** [review]

**Pull request merged**